### PR TITLE
refactor: seed data

### PR DESCRIPTION
**Pull request by emilyjablonski**
# Pull Request Template

## Issue

Addresses #1277
- [x] This change addresses the issue in full
- [ ] This change addresses only certain aspects of the issue
- [ ] This change is a dependency for another issue
- [ ] This change has a dependency from another issue

## Description

I've noticed throughout a lot of my tickets that our seed data (the three triton listings each with a different number of preferences) didn't include a wide variety of types of listings, so more than once we merged something from core to staging only to see it break something about a listing on staging that wasn't represented in the seed data listings.

This adds more seed data variety. I included:
- No preferences
- One preference
- Two preferences, same page
- FCFS
- BMR chart
- Coliseum (three preferences, separate pages, multiple rent types)
- Triton

Please let me know if you see any big part of a listing option that is not included in this list!

I also refactored the seeding part just a small amount to remove the duplication of leasing agent logic.

I had at one point planned to also base the front-end unit tests off of these new seeds during this work, but I now think that should be a separate PR.

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Prototype/POC (not to merge)
- [X] This change is a refactor/address technical debt
- [ ] This change requires a documentation update
- [ ] This change requires a SQL Script

## How Can This Be Tested/Reviewed?

Re-seed the db, checkout the listings locally :)

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have reviewed the changes in a desktop view
- [x] I have reviewed the changes in a mobile view
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules
- [x] I have assigned reviewers
- [x] I have updated the changelog to include a description of my changes

### CODE REVIEW

File: backend/core/src/seeds/listings.ts [hunk 1]
```diff
@@ -84,303 +91,2639 @@ export async function seedListing(app: INestApplicationContext, seed: ListingSee
...
+  status: ListingStatus.active,
+  waitlistCurrentSize: 0,
+  waitlistMaxSize: 3000,
+  whatToExpect: null,
+}
+
+// Seeds
```
Comment (emilyjablonski) [inline backend/core/src/seeds/listings.ts:2629]: Most everything above this is sample data. Below is where I build up the different seeds.

**Comment by pbn4** [comment]
> I had at one point planned to also base the front-end unit tests off of these new seeds during this work, but I now think that should be a separate PR.

One downside to this is that when creating changes to schema backend devs must also fix frontend related fixtures/tests which is error prone e.g. I don't have enough react/frontend experience. I'd rather separate frontend and backend completely with interfaces from autogenerated client. What do you think?

**Comment by pbn4** [comment]
@emilyjablonski What do you think about moving those seeds: 

```
export const allSeeds = [
  defaultListingSeed,
  defaultOnePreferenceSeed,
  defaultNoPreferencesSeed,
  defaultFCFS,
  defaultBMRChart,
  tritonListingSeed,
  coliseumListingSeed,
]
```

into separate files?

**Comment by pbn4** [comment]
BTW you mentioned addressing untyped tests fixtures in frontend code here https://github.com/bloom-housing/bloom/issues/1281 . What the current status of it? I know this PR does not address it, just wondering.

**Comment by emilyjablonski** [comment]
@pbn4 The seeds all share bits and pieces of the same data, so I think it would be hard to separate them across files. And I think that work should just be a separate PR, so I haven't started. When I started to look into it though it seems like most frontend tests used a typed Archer listing from the backend, so I'm not 100% sure which untyped fixtures the ticket refers to.

**Review by pbn4 — APPROVED** [review]

**Comment by pbn4** [comment]
> @pbn4 The seeds all share bits and pieces of the same data, so I think it would be hard to separate them across files, but I could also just imported the shared bits from this file? And I think that the frontend test work should just be a separate PR, so I haven't started. When I started to look into it though it seems like most frontend tests used a typed Archer listing from the backend, so I'm not 100% sure which untyped fixtures the ticket refers to.

The ticket mentions `ui-components/__tests__/fixtures/archer.json` as an example.

**Pull request merged**